### PR TITLE
User task associations #56

### LINKED ISSUE #56: Establish associations between task and user
The purpose is to provide user with access to tasks he is assign to and in the opposite way - task should have access to its participants

### PR DESCRIPTION
**Pull request by tprochniewicz**
closes #56

### CODE REVIEW

**Comment by pospieszynski** [comment]
Jak dla mnie wszystko spoko, jeszcze Krzysiek rzuci okiem.

File: spec/models/task_spec.rb [hunk 1]
```diff
@@ -36,4 +34,16 @@
       to include("Date should not be in the past.")
     end
   end
+
+  context "with properly setup many to many association user-task" do
+    it "returns empty participants list if noone is aggined" do
```
Comment (neonowy) [inline spec/models/task_spec.rb]: Typo.

File: spec/models/user_task_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  context "with valid parameters" do
+    it "assigns employee to the task" do
+      expect(UserTask.create(user: employee, task: task)).to be_valid
+    end
+
+    it "does not assigned the same person to the same task twice" do
+      expect{
+        2.times { UserTask.create(user: employee, task: task) }
+        }.to raise_error(ActiveRecord::RecordNotUnique)
```
Comment (taralalukasz) [inline spec/models/user_task_spec.rb]: too many brackets

File: spec/models/user_task_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  subject(:employee) { FactoryGirl.create(:user) }
+  let(:task) { FactoryGirl.create(:task) }
+
+  context "with valid parameters" do
+    it "assigns employee to the task" do
+      expect(UserTask.create(user: employee, task: task)).to be_valid
+    end
+
+    it "does not assigned the same person to the same task twice" do
+      expect{
```
Comment (neonowy) [inline spec/models/user_task_spec.rb]: Add space before `{`.

File: spec/models/user_task_spec.rb (hunk 2 above)
Comment (neonowy) [inline spec/models/user_task_spec.rb]: Wrong indentation.

Comment (taralalukasz) [inline spec/models/user_task_spec.rb]: wrong indentation - indent line 15 to left

**Comment by pospieszynski** [comment]
:+1: 
@gotar

**Comment by gotar** [comment]
ok good enough ;] you can merge (remember to squeeze it before (manually))

File: spec/models/task_spec.rb [hunk 4]
```diff
@@ -1,12 +1,10 @@
...
 
 RSpec.describe Task, type: :model do
-  subject(:task) do
-    Task.new(title: "Short text", 
-             description: "Long text", 
-             due_date: Date.today + 1.day )
-  end
-  
+  subject(:employee1) { FactoryGirl.create(:user) }
+  subject(:employee2) { FactoryGirl.create(:user) }
```
Comment (gotar) [inline spec/models/task_spec.rb]: they are not subjects - it's let - please change

plus it's not the best practice to name vars as foo1, foo2, foo3 - sth more descriptive will be better

File: spec/models/task_spec.rb [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
+  subject(:task) { FactoryGirl.create(:task) }
```
Comment (gotar) [inline spec/models/task_spec.rb]: it's not subject - change to let

**Pull request merged**